### ISSUE TITLE
How does openAI framework converts the ChatComplete (training data ) into model ready format (SFT)

### ISSUE THREAD

**Issue by yishangupenn** [perf-ai-example-org/openai-cookbook#231]
> **Copied from upstream:** [openai/openai-cookbook#2075](https://github.com/openai/openai-cookbook/issues/2075)
> **Original author:** @shakeel608
> **Originally created:** 2025-08-20

---

I have a question regarding the internal processing pipeline for Supervised Fine‑Tuning (SFT) when using the ChatCompletion-style JSON format for training data.

{
  "messages": [
    {
      "role": "system",
      "content": "You are an ..."
    },
    {
      "role": "user",
      "content": "Please tell me about the paper \"K205...\......""
    },
    {
      "role": "assistant",
      "content": "The paper titled \"K205R specific nanobody...\""
    }
  ]
}

How does the framework internally transform the ChatCompletion training data format into a model ready input. And how is the loss computed
